packages/navigation/src/HeaderButton.js

```diff
@@ -0,0 +1,164 @@
+import { Platform, View } from 'react-native';
+import {
+  Touchable,
+  Color,
+  StyleSheet,
+  Icon,
+  Text,
+  type StylePropType,
+} from '@kiwicom/mobile-shared';
+import {
+  type TranslationKeysType,
+  type TranslationType,
+  Translation,
+} from '@kiwicom/mobile-localization';
+
+type Props = {|
+  +onPress: () => void,
+  +onLongPress?: (React.ElementRef<typeof Touchable>) => void,
+  +children:
+    | React.Element<typeof HeaderButton.Text>
+    | React.Element<typeof Icon>,
+  +left?: boolean,
+  +style?: StylePropType,
+  +disabled?: boolean,
+|};
+
+type State = {|
+  reference: React.ElementRef<typeof Touchable> | null,
+|};
+
+type CloseModalProps = {|
+  +onPress: () => void,
+  +textId?: TranslationKeysType,
+|};
+
+export default class HeaderButton extends React.PureComponent<Props, State> {
+  static Right = (props: Props) => (
+    <HeaderButton {...props} style={[props.style, styles.right]} />
+  );
+
+  static Left = (props: Props) => (
+    <HeaderButton {...props} style={[props.style, styles.left]} />
+  );
+
+  static Text = ({
+    children,
+    disabled,
+  }: {|
+    +children: TranslationType,
+    +disabled?: boolean,
+  |}) => (
+    <Text
+      style={[styles.headerButtonText, disabled ? styles.disabledButton : null]}
+    >
+      {React.cloneElement(children, {
+        textTransform: Platform.OS === 'android' ? 'uppercase' : '',
+      })}
+    </Text>
+  );
+
+  static CloseModal = ({ onPress, textId }: CloseModalProps) => (
+    <HeaderButton.Left onPress={onPress}>
+      {Platform.OS === 'android' ? (
+        <Icon name="close" size={24} color={Color.brand} />
+      ) : (
+        <HeaderButton.Text>
+          <Translation id={textId || 'shared.button.close'} />
+        </HeaderButton.Text>
+      )}
+    </HeaderButton.Left>
+  );
+
+  static defaultProps = {
+    disabled: false,
+  };
+
+  state = {
+    reference: null,
+  };
+
+  storeReference = (reference: React.ElementRef<typeof Touchable> | null) => {
+    this.setState({ reference });
+  };
+
+  onLongPress = () => {
+    if (this.state.reference !== null && this.props.onLongPress !== undefined) {
+      this.props.onLongPress(this.state.reference);
+    }
+  };
+
+  render = () => {
+    const { disabled, onPress, style } = this.props;
+
+    return (
+      <Touchable
+        accessibilityComponentType="button"
+        accessibilityTraits="button"
+        delayPressIn={0}
+        onPress={onPress}
+        style={styles.container}
+        borderlessRipple={true}
+        onLongPress={this.onLongPress}
+        ref={this.storeReference}
+        disabled={disabled}
+      >
+        <View style={styles.container}>
+          <View style={[styles.innerContainer, style]}>
+            {this.props.children.type === HeaderButton.Text
+              ? // $FlowExpectedError: flow is having a hard time here
+                React.cloneElement(this.props.children, { disabled })
+              : this.props.children}
```

Comment:
I am also having a hard time here, what and why is the text disabled?

Reply:
So `Button` can be `disabled` but you also need to pass it to `Text` because then it has a special style (disabled color right).

Reply:
Right, I understand